back/engines/commercial/public_api/spec/acceptance/v2/idea_phases_spec.rb

```diff
@@ -36,7 +36,7 @@
     let(:ideas) { create_list(:idea, 2, project: project) }
     let!(:idea_phases) do
       2.times do |index|
-        create(:ideas_phase, idea: ideas[index], phase: project.phases[index])
```

Comment:
Many specs were changed, because:
1. The policy for updating ideas is now using the idea permissions service, which always requires the idea to be in the current phase, regardless of the action. After some hesitation, I consider it a bug that it was allowed to update an input that is not in the current phase (moderators can always update though). For example commenting, reacting, reacting on comments are all forbidden when the idea is not in the current phase.
2. This caused many specs to fail because the ideas created by the factories where not in the current phase. So I changed the factories to always put the created inputs in the current phase (or at least in all compatible project phases).
3. This caused some specs to fail, when trying to add an idea to a phase it was already assigned to. Those specs were easier to fix, and most specs changes in this PR are because of this.

Reply:
This comment is now outdated. I think it's fine to keep the spec changes to save time (spent on reverting these changes) and as it might be a more "typical" setup.